src/main/java/net/pms/util/Rational.java

```diff
@@ -0,0 +1,1492 @@
+	 *
+	 * @return This {@link Rational} converted to an {@code int}.
+	 */
+	@Override
+	public int intValue() {
+		return new BigDecimal(reducedNumerator).divide(new BigDecimal(reducedDenominator), RoundingMode.DOWN).intValue();
+	}
+
+	/**
+	 *
+	 * Converts this {@link Rational} to a {@code long}. This conversion is
+	 * analogous to the <i>narrowing primitive conversion</i> from
+	 * {@code double} to {@code long} as defined in section 5.1.3 of <cite>The
+	 * Java&trade; Language Specification</cite>: any fractional part of this
+	 * {@link Rational} will be discarded, and if the resulting "
+	 * {@link BigInteger}" is too big to fit in a {@code long}, only the
+	 * low-order 64 bits are returned.
+	 * <p>
+	 * Note that this conversion can lose information about the overall
+	 * magnitude and precision of this {@link Rational} value as well as return
+	 * a result with the opposite sign.
+	 *
+	 * @return This {@link Rational} converted to a {@code long}.
+	 */
+	@Override
+	public long longValue() {
+		return new BigDecimal(reducedNumerator).divide(new BigDecimal(reducedDenominator), RoundingMode.DOWN).longValue();
+	}
+
+	/**
+	 * Converts this {@link Rational} to a {@link BigInteger}. This conversion
+	 * is analogous to the <i>narrowing primitive conversion</i> from
+	 * {@code double} to {@code long} as defined in section 5.1.3 of <cite>The
+	 * Java&trade; Language Specification</cite>: any fractional part of this
+	 * {@link Rational} will be discarded.
+	 *
+	 * @return This {@link Rational} converted to a {@link BigInteger}.
+	 */
+	public BigInteger bigIntegerValue() {
+		return new BigDecimal(reducedNumerator).divide(new BigDecimal(reducedDenominator), RoundingMode.DOWN).toBigInteger();
+	}
+
+	/**
+	 * Converts this {@link Rational} to a {@code float}. This conversion is
+	 * similar to the <i>narrowing primitive conversion</i> from {@code double}
+	 * to {@code float} as defined in section 5.1.3 of <cite>The Java&trade;
+	 * Language Specification</cite>: if this {@link Rational} has too great a
+	 * magnitude to represent as a {@code float}, it will be converted to
+	 * {@link Float#NEGATIVE_INFINITY} or {@link Float#POSITIVE_INFINITY} as
+	 * appropriate.
+	 * <p>
+	 * Note that even when the return value is finite, this conversion can lose
+	 * information about the precision of the {@link Rational} value.
+	 *
+	 * @return This {@link Rational} converted to a {@code float}.
+	 */
+	@Override
+	public float floatValue() {
+		return new BigDecimal(reducedNumerator).divide(new BigDecimal(reducedDenominator), MathContext.DECIMAL32).floatValue();
+	}
+
+	/**
+	 * Converts this {@link Rational} to a {@code double}. This conversion is
+	 * similar to the <i>narrowing primitive conversion</i> from {@code double}
+	 * to {@code float} as defined in section 5.1.3 of <cite>The Java&trade;
+	 * Language Specification</cite>: if this {@link Rational} has too great a
+	 * magnitude to represent as a {@code double}, it will be converted to
```

Comment:
@Sami32 I guess one could say "has a too great magnitude" as well, but I copied and modified this text from the Java API so the wording is from Oracle.